src/stores/video.ts

```diff
@@ -141,6 +141,71 @@ export const useVideoStore = defineStore('video', () => {
     alertStore.pushAlert(new Alert(AlertLevel.Success, `Stopped recording stream ${streamName}.`))
   }
 
+  const extractThumbnail = (): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const video = document.getElementById('mainDisplayStream') as HTMLVideoElement | null
```

Comment:
There's no need here to get the element by ID, as it's already binded on the template via the "videoElement" ref.

You can read more about it [here](https://vuejs.org/guide/essentials/template-refs#ref-on-component).